src/string_object.cpp

```diff
@@ -1085,7 +1085,9 @@ Value StringObject::crypt(Env *env, Value salt) {
     if (strlen(salt_str->c_str()) < 2)
         env->raise("ArgumentError", "salt too short (need >=2 bytes)");
 
-    return new StringObject { ::crypt(c_str(), salt_str->c_str()) };
+    auto crypted = ::crypt(c_str(), salt_str->c_str());
+    assert(crypted);
```

Comment:
According to `man crypt(3)`: `set errno when they fail.`, so this could be turned into:
```ruby
if (!crypted)
  env->raise_errno();
```
This might give a bit more information about the source of the error.